swift/Lottie.swift

```diff
@@ -0,0 +1,152 @@
+    ///   - path: The file path of the Lottie animation.
+    public convenience init(path: String) throws {
+        let animation = tvg_animation_new()
+        let picture = tvg_animation_get_picture(animation)
+
+        guard tvg_picture_load(picture, path) == TVG_RESULT_SUCCESS else {
+            throw LottieError.failedToLoadLottieFromPath
+        }
+
+        try self.init(animation: animation)
+    }
+
+    /// Initializes a new Lottie instance from a string representing the animation.
+    /// - Parameters:
+    ///   - string: The string representing the Lottie animation.
+    public convenience init(string: String) throws {
+        let animation = tvg_animation_new()
+        let picture = tvg_animation_get_picture(animation)
+
+        guard let cString = string.cString(using: .utf8),
+              tvg_picture_load_data(picture, cString, UInt32(cString.count), "lottie", "", false) == TVG_RESULT_SUCCESS
+        else {
+            throw LottieError.failedToLoadLottieFromString
+        }
+
+        try self.init(animation: animation)
+    }
+
+    /// Initializes the Lottie instance with frame count and duration extracted from the animation pointer.
+    private init(animation: Animation?) throws {
+        var numberOfFrames: Float = 0
+        tvg_animation_get_total_frame(animation, &numberOfFrames)
+
+        var duration: Float = 0
+        tvg_animation_get_duration(animation, &duration)
+
+        self.animation = animation
+        self.numberOfFrames = Int(numberOfFrames)
+        self.duration = CMTime(seconds: Double(duration), preferredTimescale: 600)
+    }
+
+    /// Renders a Lottie frame into a buffer.
+    /// - Parameters:
+    ///   - index: The index of the frame to render.
+    ///   - buffer: The buffer to render the frame into.
+    ///   - stride: The stride of the buffer.
+    ///   - size: The desired size of the rendered frame.
+    ///   - crop: Optional rectangle to crop the rendered frame.
+    public func render(frameAt index: Int, into buffer: Buffer, stride: Int, size: CGSize, crop: CGRect? = nil) throws {
+        guard index < numberOfFrames, index >= 0 else {
+            throw LottieError.frameIndexOutOfBounds
+        }
+
+        if canvas == nil {
+            canvas = try createCanvas(with: buffer, stride: stride, size: size)
+        }
+
+        prepareCanvasForRendering(frameIndex: index, size: size, crop: crop)
+        try renderCanvas()
+    }
+
+    /// Creates a new canvas for rendering, initializing the TVG engine and setting the target buffer.
+    private func createCanvas(with buffer: Buffer, stride: Int, size: CGSize) throws -> Canvas? {
+        guard tvg_engine_init(TVG_ENGINE_SW, 4) == TVG_RESULT_SUCCESS else {
+            throw LottieError.failedToInitialiseTvgEngine
+        }
+
+        let canvas = tvg_swcanvas_create()
+        tvg_swcanvas_set_target(canvas, buffer, UInt32(stride), UInt32(size.width), UInt32(size.height), TVG_COLORSPACE_ARGB8888)
```

Comment:
Looking at the code and the documentation here, I think the more important one here is `tvg_canvas_push`. As discussed, this can throw a `TVG_RESULT_INSUFFICIENT_CONDITION`, and is important step to begin preprocessing. I'll add a `guard` around that call.